src/rpc/rawtransaction_util.cpp

```diff
@@ -355,28 +348,31 @@ CMutableTransaction ConstructTransaction(const UniValue& inputs_in, const UniVal
                 CPubKey blind_pub;
                 if (IsBlindDestination(destination)) {
                     blind_pub = GetDestinationBlindingKey(destination);
-                    if (!output_pubkeys_out) {
+                    if (!outputs_aux) {
                         // Only use the pubkey-in-nonce hack if the caller is not getting the pubkeys the nice way.
                         out.nNonce.vchCommitment = std::vector<unsigned char>(blind_pub.begin(), blind_pub.end());
                     }
                 }
-                if (output_pubkeys_out) {
-                    output_pubkeys_out->push_back(blind_pub);
-                }
+                psbt_out.m_blinding_pubkey = blind_pub;
             }
         }
         if (is_fee) {
             fee_out = out;
         } else {
             rawTx.vout.push_back(out);
+            psbt_outs.push_back(psbt_out);
         }
     }
 
     // Add fee output in the end.
     if (!fee_out.nValue.IsNull() && fee_out.nValue.GetAmount() > 0) {
         rawTx.vout.push_back(fee_out);
-        if (output_pubkeys_out) {
-            output_pubkeys_out->push_back(CPubKey());
+        psbt_outs.emplace_back(2);
```

Comment:
Can you add a comment here (and several lines above at `PSBTOutput psbt_out(2);`) that "2" represents a dummy PSBT output with version 2?

Reply:
Done.